managers/printer_manager.py

```diff
@@ -0,0 +1,127 @@
+"""
+module: manager
+"""
+from abc import ABC
+
+from models.printer import Printer
+
+
+class PrinterManager(ABC):
+    """
+    Клас, що управляє принтерами.
+    """
+
+    def __init__(self):
+        """
+        Ініціалізує об'єкт класу PrinterManager.
+        """
+        self.printers = []
+
+    def add_printer(self, printer):
+        """
+        Додає принтер до списку.
+
+        Arguments:
+        - printer: об'єкт принтера, який додається до списку.
+        """
+        self.printers.append(printer)
+
+    def remove_printer(self, printer):
+        """
+        Видаляє принтер зі списку.
+
+        Arguments:
+        - printer: об'єкт принтера, який видаляється зі списку.
+        """
+        self.printers.remove(printer)
+
+    def print_all(self, pages):
+        """
+        Друкує задану кількість сторінок на всіх принтерах у списку.
+
+        Arguments:
+        - pages: кількість сторінок, які необхідно роздрукувати.
+        """
+        for printer in self.printers:
+            printer.print(pages)
+
+    def __len__(self):
+        """
+        Повертає довжину списку принтерів.
+
+        Returns:
+        - довжина списку принтерів.
+        """
+        return len(self.printers)
+
+    def __getitem__(self, index):
+        """
+        Повертає принтер за індексом.
+
+        Arguments:
+        - index: індекс принтера у списку.
+
+        Returns:
+        - об'єкт принтера за заданим індексом.
+        """
+        return self.printers[index]
+
+    def __iter__(self):
+        """
+        Повертає ітератор для ітерації по списку принтерів.
+
+        Returns:
+        - ітератор по списку принтерів.
+        """
+        return iter(self.printers)
+
+    def get_results(self):
+        """
+        Повертає список результатів виклику методу do_something() для всіх принтерів.
+
+        Returns:
+        - список результатів виклику методу do_something() для всіх принтерів.
+        """
+        return [printer.do_something() for printer in self.printers]
+
+    def enumerate_objects(self):
+        """
+        Повертає список пар (порядковий номер, принтер) для всіх принтерів.
+
+        Returns:
+        - список пар (порядковий номер, принтер) для всіх принтерів.
+        """
+        return list(enumerate(self.printers))
```

Comment:
no need to cast it to list immideately. If you want it as list, you can cast it right there. In most cases, it would be more useful as simple iterator